media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt

```diff
@@ -136,43 +137,51 @@ class AudioPlayerService : DaggerService() {
      * * [Playing]
      * * [Paused]
      * * [Stopped]
+     *
+     * To switch states, send a [PlayBackAction] to [audioControl].
      */
-    sealed class PlayBackState(val articAudioFile: ArticAudioFile) {
-        class Playing(articAudioFile: ArticAudioFile) : PlayBackState(articAudioFile)
-        class Paused(articAudioFile: ArticAudioFile) : PlayBackState(articAudioFile)
-        class Stopped(articAudioFile: ArticAudioFile) : PlayBackState(articAudioFile)
+    sealed class PlayBackState(val audio: AudioTranslation) {
+        class Playing(audio: AudioTranslation) : PlayBackState(audio)
+        class Paused(audio: AudioTranslation) : PlayBackState(audio)
+        class Stopped(audio: AudioTranslation) : PlayBackState(audio)
     }
 
     private val binder: Binder = AudioPlayerServiceBinder()
     private lateinit var playerNotificationManager: PlayerNotificationManager
+
+
     var articObject: ArticObject? = null
+        private set
+    var preferredAudio: AudioTranslation? = null
 
     private val audioControl: Subject<PlayBackAction> = BehaviorSubject.create()
     val audioPlayBackStatus: Subject<PlayBackState> = BehaviorSubject.create()
-    val currentTrack: Subject<ArticAudioFile> = BehaviorSubject.create()
+    val currentTrack: Subject<AudioTranslation> = BehaviorSubject.create()
 
     val disposeBag = DisposeBag()
 
     @Inject
     lateinit var analyticsTracker: AnalyticsTracker
+    @Inject
+    lateinit var languageSelector: LanguageSelector
```

Comment:
move to constructor

Reply:
Both of the fields, right? Shouldn't be any trouble.

Reply:
...turns out `Service` objects can't have constructor params. I forgot that. Leaving these fields as is with an extra comment explaining why.